### PR TITLE
Suggest features implementation

### PR DESCRIPTION
**Pull request by jotare**
### Description
- Add `features` to `SuggestRequest` so we can select which search to perform in the Rust side
- Remove unused `INTENT` suggest feature
- Improve Rust integration test performance by tweaking `backoff` times
- Replace Python `/suggest` integration tests for faster Rust ones

### How was this PR tested?
- New Rust integration tests
- Python tests

### CODE REVIEW

File: nucliadb_node/src/shards/shard_reader.rs [hunk 1]
```diff
@@ -258,53 +258,85 @@ impl ShardReader {
...
-            .map(|prefix| RelationSearchRequest {
-                shard_id: String::default(),
-                prefix: Some(prefix),
-                ..Default::default()
-            })
-            .map(|request| relations_reader_service.search(&request));
 
-        let relation_task = move || relations.collect::<Vec<_>>();
-        let paragraph_task = move || paragraph_reader_service.suggest(&request);
+        let suggest_paragraphs_task = suggest_paragraphs.then(|| {
```
Comment (hermeGarcia) [inline nucliadb_node/src/shards/shard_reader.rs:272]: Cool use of `then`!

**Review by hermeGarcia — APPROVED** [review]
Rust code looks good to me!

File: nucliadb/nucliadb/search/search/query.py [hunk 2]
```diff
@@ -229,6 +235,8 @@ async def suggest_query_to_pb(
     if range_modification_end is not None:
         request.timestamps.to_modified.FromDatetime(range_modification_end)
 
+    print("Py Suggest Request:")
+    print(request)
```
Comment (lferran) [inline nucliadb/nucliadb/search/search/query.py:239]: ups! delete

**Review by lferran — APPROVED** [review]

File: nucliadb/nucliadb/search/search/query.py (hunk 2 above)
Comment (jotare) [reply nucliadb/nucliadb/search/search/query.py:239]: Thanks!

**Pull request merged**